hs_core/views/resource_rest_api.py

```diff
@@ -0,0 +1,622 @@
+    :param extract_metadata: (optional) to indicate if metadata to be extracted from the file being added
+    :return: id of the resource and name of the file added
+    :rtype: json string of format: {'resource_id':pk, 'file_name': name of the file added}
+    REST URL: hsapi/resource/{pk}/files/{filename}
+    HTTP method: DELETE
+    HTTP status code (on success): 200
+    :type pk: str
+    :type filename: str
+    :param pk: resource id
+    :param filename: name of the file to delete
+    :return: id of the resource and name of the file
+    :rtype: json string of format: {'resource_id':pk, 'file_name': name of the file deleted}
+    :raises:
+    &nbsp;&nbsp;&nbsp;NotFound: return json format: {'detail': 'No resource was found for resource id':pk}
+    &nbsp;&nbsp;&nbsp;HTTP status code: 404
+    &nbsp;&nbsp;&nbsp;PermissionDenied: return json format: {'detail': 'You do not have permission to perform this action.'}
+    &nbsp;&nbsp;&nbsp;HTTP status code:403
+    &nbsp;&nbsp;&nbsp;ValidationError: return json format: {'parameter-1':['error message-1'], 'parameter-2': ['error message-2'], .. }
+    &nbsp;&nbsp;&nbsp;HTTP status code:400
+    """
+
+    __rest_api_response = utils.HSRestAPIResponse()
+
+    @property
+    def allowed_methods(self):
+        return ['GET', 'POST', 'DELETE']
+
+    def get(self, request, pk, filename):
+        # check resource exists for the provided resource id (pk)
+        #utils.get_resource(resource_id=pk)
+
+        # check the user is authorized for this action and if the resource exists for the given resource id (pk)
+        utils.authorize(request, pk, view=True)
+        try:
+            f = hydroshare.get_resource_file(pk, filename)
+        except:
+            err_msg = 'File with file name {file_name} does not exist for resource with resource id ' \
+                      '{res_id}'.format(file_name=filename, res_id=pk)
+            raise NotFound(detail=err_msg)
+
+        # redirects to django_irods/views.download function
+        return HttpResponseRedirect(f.url)
+
+    def post(self, request, pk):
+        #resource = utils.get_resource(resource_id=pk)
+        # check the user is authorized for this action and if the resource exists for the given resource id (pk)
+        resource, _, _ = utils.authorize(request, pk, edit=True)
+        resource_files = request.FILES.values()
+        if len(resource_files) == 0:
+            error_msg = {'missing_file': 'No file was provided to add to the resource.'}
+            raise ValidationError(detail=error_msg)
+
+        # TODO: user utils.validate_resource_file_size() here
+        valid = hydroshare.check_resource_files(resource_files)
+        if not valid:
+            error_msg = {
+                'file_size_error': 'The resource file is larger than the supported size limit %s. '
+                                    % file_size_limit_for_display
+            }
+            raise ValidationError(detail=error_msg)
```

Comment:
This should really be enforced at the web server level somehow... here we are storing the entire file in memory and THEN checking the file size. That's asking for trobule.

This is also a separate issue because it already exists in the UI, too, and we need to solve for both cases. Let's file a follow up ticket for it.